test/e2e/declare_consumer.test.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from "chai"
+import { connect } from "../../src"
+import { Message } from "../../src/producer"
+import { Offset } from "../../src/requests/subscribe_request"
+import { Rabbit } from "../support/rabbit"
+import { eventually, expectToThrowAsync } from "../support/util"
+
+describe("declare consumer", () => {
+  const rabbit = new Rabbit()
+  const testStreamName = "test-stream"
+  const nonExistingStream = "not-the-test-stream"
+
+  beforeEach(async () => {
+    await rabbit.createStream(testStreamName)
+  })
+
+  afterEach(async () => {
+    await rabbit.deleteStream(testStreamName)
+  })
+
+  it("declaring a consumer on an existing stream - the consumer should handle the message", async () => {
+    const connection = await connect({
+      hostname: "localhost",
+      port: 5552,
+      username: "rabbit",
+      password: "rabbit",
+      vhost: "/",
+      frameMax: 0,
+      heartbeat: 0,
+    })
```

Comment:
I suggest to move the connection in beforeEach (and its close in afterEach) so if an test fail the connection is closed